scipy/stats/_stats_py.py

```diff
@@ -738,11 +749,12 @@ def tmin(a, lowerlimit=None, axis=0, inclusive=True, nan_policy='propagate'):
         am = ma.masked_invalid(am)
 
     res = ma.minimum.reduce(am, axis).data
```

Comment:
I wonder if `min` didn't exist when this was written?

Hmm there are some issues:
```python3
stats.tmin([[-2, -2, -2], [1, 2, 3]], axis=1, lowerlimit=-1)  # array([2147483647,          1])
stats.tmin([[-2, -2, -2.], [1, 2, 3]], axis=1, lowerlimit=-1)  # array([inf,  1.])
```

Again, could be a separate PR, but these functions should be so simple that I'd like to clean them up if you're willing.

Reply:
I think max wasn't used since it returns scalars if axis isn't mentioned so it needs to be handles by a separate if statement. `ma.maximum.reduce` always returns masked arrays. Changed it to use `ma.max`, let me know if it looks simpler/more readable.

Reply:
I also noticed an error in `stats.tmin` and `stats.tmax` while working on this. Currently, if a slice gets completely masked, it returns a garbage value instead of something more sensible like a NaN.

```python
In [1]: import numpy as np

In [2]: from scipy import stats
x
In [3]: x = np.arange(64).reshape(8,8)

In [4]: stats.tmin(x, lowerlimit=8, axis=1)
Out[4]: 
array([9223372036854775807,                   8,                  16,
                        24,                  32,                  40,
                        48,                  56])
```

I have fixed it here and added a test for it.